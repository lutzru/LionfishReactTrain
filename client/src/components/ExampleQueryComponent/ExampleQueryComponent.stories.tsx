import { action } from '@storybook/addon-actions'
import { withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import ExampleQueryComponent, {
    ExampleQueryComponentProps,
} from 'components/ExampleQueryComponent/ExampleQueryComponent'
import * as React from 'react'
import 'utils/ i18n'

const stories = storiesOf('Example Component', module)

const props: ExampleQueryComponentProps = {
    loadLastSavedCount: action('loadLastSavedCount'),
    saveCount: action('saveCount'),
}

stories.addDecorator(withKnobs)
stories.add('default view', () => <ExampleQueryComponent {...props} />)
