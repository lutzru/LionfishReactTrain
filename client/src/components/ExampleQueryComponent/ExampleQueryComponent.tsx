import { Button, Grid, Typography } from '@material-ui/core'
import React from 'react'
import { useTranslation } from 'react-i18next'
import TestIDs from 'utils/TestIDs'

export interface ExampleQueryComponentProps {
    saveCount: () => void
    loadLastSavedCount: () => void
}

const ExampleQueryComponent = (props: ExampleQueryComponentProps): JSX.Element => {
    const { loadLastSavedCount, saveCount } = props
    const { t } = useTranslation()

    return (
        <Grid item container xs={12} spacing={3}>
            <Grid item xs={12} sm>
                <Button
                    {...TestIDs.GET_EXAMPLE('EXAMPLE_SAVE')}
                    onClick={saveCount}
                    fullWidth
                    variant={'outlined'}
                    color={'primary'}
                >
                    <Typography className={'ExampleText'}>{t('save')}</Typography>
                </Button>
            </Grid>
            <Grid item xs={12} sm>
                <Button
                    {...TestIDs.GET_EXAMPLE('EXAMPLE_LOAD')}
                    onClick={loadLastSavedCount}
                    fullWidth
                    variant={'outlined'}
                    color={'primary'}
                >
                    <Typography className={'ExampleText'}>{t('load')}</Typography>
                </Button>
            </Grid>
        </Grid>
    )
}

export default ExampleQueryComponent
