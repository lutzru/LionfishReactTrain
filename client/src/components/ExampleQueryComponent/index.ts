import ExampleQueryComponent from './ExampleQueryComponent'

export default ExampleQueryComponent
