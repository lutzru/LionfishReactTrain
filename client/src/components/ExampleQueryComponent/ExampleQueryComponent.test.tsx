import { Button } from '@material-ui/core'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import ExampleQueryComponent, { ExampleQueryComponentProps } from './ExampleQueryComponent'

jest.mock('react-i18next', () => ({
    useTranslation: (): unknown => ({ t: (key: string): string => key }),
}))

const props: ExampleQueryComponentProps = {
    loadLastSavedCount: jest.fn(),
    saveCount: jest.fn(),
}

describe('ExampleQueryComponent Component', () => {
    it('should render correctly with react-test-renderer', () => {
        const component = renderer.create(<ExampleQueryComponent {...props} />)

        expect(component).toMatchSnapshot()
    })
    it('should render correctly with enzyme', () => {
        const component = shallow(<ExampleQueryComponent {...props} />)

        expect(toJson(component)).toMatchSnapshot()
    })
    it('should handle a click correctly', () => {
        const onSaveMock = jest.fn()
        const onLoadMock = jest.fn()
        const component = shallow(<ExampleQueryComponent loadLastSavedCount={onLoadMock} saveCount={onSaveMock} />)

        component
            .find(Button)
            .at(0)
            .simulate('click', { currentTarget: { value: 0 } })
        expect(onSaveMock).toHaveBeenCalledTimes(1)
        component
            .find(Button)
            .at(1)
            .simulate('click', { currentTarget: { value: 0 } })
        expect(onLoadMock).toHaveBeenCalledTimes(1)
    })
})
