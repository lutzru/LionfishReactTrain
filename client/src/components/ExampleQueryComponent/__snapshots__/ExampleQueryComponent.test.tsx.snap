// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleQueryComponent Component should render correctly with enzyme 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  item={true}
  spacing={3}
  xs={12}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      data-testid="EXAMPLE_PAGE_EXAMPLE_SAVE"
      fullWidth={true}
      onClick={[MockFunction]}
      variant="outlined"
    >
      <WithStyles(ForwardRef(Typography))
        className="ExampleText"
      >
        save
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    sm={true}
    xs={12}
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      data-testid="EXAMPLE_PAGE_EXAMPLE_LOAD"
      fullWidth={true}
      onClick={[MockFunction]}
      variant="outlined"
    >
      <WithStyles(ForwardRef(Typography))
        className="ExampleText"
      >
        load
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`ExampleQueryComponent Component should render correctly with react-test-renderer 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item MuiGrid-grid-xs-12"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
      data-testid="EXAMPLE_PAGE_EXAMPLE_SAVE"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root ExampleText MuiTypography-body1"
        >
          save
        </p>
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
      data-testid="EXAMPLE_PAGE_EXAMPLE_LOAD"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root ExampleText MuiTypography-body1"
        >
          load
        </p>
      </span>
    </button>
  </div>
</div>
`;
